[BUG] Fix failing tests in tabular

tests fail on main:

```
=========================== short test summary info ============================
FAILED tests/tabular/checks/model_evaluation/simple_model_comparison_test.py::test_regression_tree - IndexError: invalid index to scalar variable.
FAILED tests/tabular/checks/model_evaluation/simple_model_comparison_test.py::test_regression_tree_random_state - IndexError: invalid index to scalar variable.
FAILED tests/tabular/checks/model_evaluation/simple_model_comparison_test.py::test_regression_tree_max_depth - IndexError: invalid index to scalar variable.
FAILED tests/tabular/suites/test_suites.py::test_kiss_dataset_suite - AssertionError: 
Expected: a sequence containing items matching (an instance of DeepchecksBaseError)
     but: was <[NotEnoughSamplesError("Not enough samples to calculate drift score. Minimum 10 samples required. Note that for numerical labels, None values do not count as samples.Use the 'min_samples' parameter to change this requirement."), DeepchecksProcessError('WeakSegmentsPerformance was unable to train an error model to find weak segments. Try increasing n_samples or supply additional features.'), DeepchecksValueError('Label observed values [1, 3] contain values that are not found in the model classes: [1, 2, 5].'), ModelValidationError('Check is irrelevant for classification tasks'), ModelValidationError('Check is irrelevant for classification tasks'), IndexError('invalid index to scalar variable.'), IndexError('invalid index to scalar variable.'), DatasetValidationError('Dataset does not contain a datetime'), DatasetValidationError('Dataset does not contain a datetime'), DatasetValidationError('Dataset does not contain an index'), NotEnoughSamplesError("Not enough samples to calculate drift score. Minimum 10 samples required. Note that for numerical columns, None values do not count as samples.Use the 'min_samples' parameter to change this requirement."), NotEnoughSamplesError("Not enough samples to calculate drift score. Minimum 10 samples required. Note that for numerical labels, None values do not count as samples.Use the 'min_samples' parameter to change this requirement."), NotEnoughSamplesError('There are not enough samples to run this check, found only 4 samples.'), NotEnoughSamplesError('There are not enough samples to run this check, found only 2 samples.'), DatasetValidationError('Dataset does not contain an index or a datetime'), DatasetValidationError('Dataset does not contain an index or a datetime')]>
FAILED tests/tabular/suites/test_suites.py::test_weird_classification_suite - AssertionError: 
Expected: a sequence containing items matching (an instance of DeepchecksBaseError)
     but: was <[ModelValidationError('Check is irrelevant for classification tasks'), ModelValidationError('Check is irrelevant for classification tasks'), IndexError('invalid index to scalar variable.'), IndexError('invalid index to scalar variable.'), DatasetValidationError('Dataset does not contain a datetime'), DatasetValidationError('Dataset does not contain a datetime')]>
FAILED tests/tabular/suites/test_suites.py::test_weird_regression_suite - AssertionError: 
Expected: a sequence containing items matching (an instance of DeepchecksBaseError)
     but: was <[ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), IndexError('invalid index to scalar variable.'), IndexError('invalid index to scalar variable.'), ModelValidationError('Check is irrelevant for regression tasks'), DatasetValidationError('Dataset does not contain a datetime'), DatasetValidationError('Dataset does not contain a datetime'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks')]>
FAILED tests/tabular/suites/test_suites.py::test_missing_test_classes_suite - AssertionError: 
Expected: a sequence containing items matching (an instance of DeepchecksBaseError)
     but: was <[ModelValidationError('Check is irrelevant for classification tasks'), ModelValidationError('Check is irrelevant for classification tasks'), IndexError('invalid index to scalar variable.'), IndexError('invalid index to scalar variable.'), DatasetValidationError('Dataset does not contain a datetime'), DatasetValidationError('Dataset does not contain a datetime')]>
FAILED tests/tabular/suites/test_suites.py::test_diabetes_dataset_suite - AssertionError: 
Expected: a sequence containing items matching (an instance of DeepchecksBaseError)
     but: was <[ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), IndexError('invalid index to scalar variable.'), IndexError('invalid index to scalar variable.'), ModelValidationError('Check is irrelevant for regression tasks'), DatasetValidationError('Dataset does not contain a datetime'), DatasetValidationError('Dataset does not contain a datetime'), DatasetValidationError('Dataset does not contain an index'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), DatasetValidationError('Dataset does not contain an index or a datetime'), DatasetValidationError('Dataset does not contain an index or a datetime')]>
FAILED tests/tabular/suites/test_suites.py::test_generic_boost - AssertionError: 
Expected: a sequence containing items matching (an instance of DeepchecksBaseError)
     but: was <[ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), IndexError('invalid index to scalar variable.'), IndexError('invalid index to scalar variable.'), ModelValidationError('Check is irrelevant for regression tasks'), DatasetValidationError('Dataset does not contain a datetime'), DatasetValidationError('Dataset does not contain a datetime'), DatasetValidationError('Dataset does not contain an index'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), DatasetValidationError('Dataset does not contain an index or a datetime'), DatasetValidationError('Dataset does not contain an index or a datetime')]>
FAILED tests/tabular/suites/test_suites.py::test_generic_custom - AssertionError: 
Expected: a sequence containing items matching (an instance of DeepchecksBaseError)
     but: was <[ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), IndexError('invalid index to scalar variable.'), IndexError('invalid index to scalar variable.'), ModelValidationError("Check is relevant for Boosting models of type ('AdaBoostClassifier', 'GradientBoostingClassifier', 'LGBMClassifier', 'XGBClassifier', 'CatBoostClassifier', 'AdaBoostRegressor', 'GradientBoostingRegressor', 'LGBMRegressor', 'XGBRegressor', 'CatBoostRegressor'), but received model of type MyModel"), ModelValidationError('Check is irrelevant for regression tasks'), DatasetValidationError('Dataset does not contain a datetime'), DatasetValidationError('Dataset does not contain a datetime'), DatasetValidationError('Dataset does not contain an index'), ModelValidationError('Check is irrelevant for regression tasks'), ModelValidationError('Check is irrelevant for regression tasks'), DatasetValidationError('Dataset does not contain an index or a datetime'), DatasetValidationError('Dataset does not contain an index or a datetime')]>
==== 10 failed, 1264 passed, 2 skipped, 7433 warnings in 1466.45s (0:24:26) ====
```

--- a/deepchecks/vision/utils/image_functions.py
+++ b/deepchecks/vision/utils/image_functions.py
@@ -268,8 +268,8 @@ def prepare_thumbnail(
         # Takes the minimum factor in order for the image to not exceed the size in either width or height
         factor = min(width_factor, height_factor)
         size = (int(image.size[0] * factor), int(image.size[1] * factor))
-        # Resize the image by Resampling.LANCZOS
-        image = image.resize(size, pilimage.Resampling.LANCZOS)
+        # Resize the image by Image.LANCZOS
+        image = image.resize(size, pilimage.LANCZOS)
     else:
         image = ensure_image(image, copy=False)
 

--- a/deepchecks/vision/utils/image_functions.py
+++ b/deepchecks/vision/utils/image_functions.py
@@ -225,7 +225,7 @@ def get_font_with_size(text, desired_width):
     desired_width = max(100, desired_width)
     jump_size = 100
     while jump_size > 1:
-        if font.getsize(text)[0] < desired_width:
+        if font.size(text)[0] < desired_width:
             font_size += jump_size
         else:
             jump_size = jump_size // 2

--- a/deepchecks/vision/utils/image_functions.py
+++ b/deepchecks/vision/utils/image_functions.py
@@ -225,7 +225,9 @@ def get_font_with_size(text, desired_width):
     desired_width = max(100, desired_width)
     jump_size = 100
     while jump_size > 1:
-        if font.size(text)[0] < desired_width:
+        left, _, right, _ = font.getbbox(text)
+        width = right - left
+        if width < desired_width:
             font_size += jump_size
         else:
             jump_size = jump_size // 2

--- a/deepchecks/vision/utils/image_functions.py
+++ b/deepchecks/vision/utils/image_functions.py
@@ -266,8 +266,8 @@ def prepare_thumbnail(
         # Takes the minimum factor in order for the image to not exceed the size in either width or height
         factor = min(width_factor, height_factor)
         size = (int(image.size[0] * factor), int(image.size[1] * factor))
-        # Resize the image
-        image = image.resize(size, pilimage.ANTIALIAS)
+        # Resize the image by Resampling.LANCZOS
+        image = image.resize(size, pilimage.Resampling.LANCZOS)
     else:
         image = ensure_image(image, copy=False)
 

--- a/requirements/dev-requirements.txt
+++ b/requirements/dev-requirements.txt
@@ -37,7 +37,7 @@ opencv-python>=4.1.2
 Pillow>=7.1.2
 PyYAML>=5.3.1
 requests>=2.23.0
-scipy>=1.4.1
+scipy>=1.4.1, <=1.10.1;
 tqdm>=4.41.0
 seaborn>=0.11.0
 wandb>=0.12.15,<0.13.0

--- a/requirements/dev-requirements.txt
+++ b/requirements/dev-requirements.txt
@@ -34,7 +34,7 @@ kaleido  # optional dependency, used by plotly to transform figures into images
 matplotlib>=3.2.2
 numpy>=1.18.5
 opencv-python>=4.1.2
-Pillow>=7.1.2
+Pillow>=7.1.2, <=9.5.0
 PyYAML>=5.3.1
 requests>=2.23.0
 scipy>=1.4.1, <=1.10.1

--- a/requirements/dev-requirements.txt
+++ b/requirements/dev-requirements.txt
@@ -37,6 +37,8 @@ opencv-python>=4.1.2
 Pillow>=7.1.2
 PyYAML>=5.3.1
 requests>=2.23.0
+# Remove the <=1.10.1 dependency below once sklearn's issue is fixed. The higher version causes
+# issues with sklear's _most_frequent() function using scipy's mode() function
 scipy>=1.4.1, <=1.10.1
 tqdm>=4.41.0
 seaborn>=0.11.0

--- a/requirements/dev-requirements.txt
+++ b/requirements/dev-requirements.txt
@@ -34,7 +34,7 @@ kaleido  # optional dependency, used by plotly to transform figures into images
 matplotlib>=3.2.2
 numpy>=1.18.5
 opencv-python>=4.1.2
-Pillow>=7.1.2, <=9.5.0
+Pillow>=7.1.2
 PyYAML>=5.3.1
 requests>=2.23.0
 scipy>=1.4.1, <=1.10.1

--- a/requirements/dev-requirements.txt
+++ b/requirements/dev-requirements.txt
@@ -37,7 +37,7 @@ opencv-python>=4.1.2
 Pillow>=7.1.2
 PyYAML>=5.3.1
 requests>=2.23.0
-scipy>=1.4.1, <=1.10.1;
+scipy>=1.4.1, <=1.10.1
 tqdm>=4.41.0
 seaborn>=0.11.0
 wandb>=0.12.15,<0.13.0

--- a/requirements/dev-requirements.txt
+++ b/requirements/dev-requirements.txt
@@ -38,7 +38,7 @@ Pillow>=7.1.2
 PyYAML>=5.3.1
 requests>=2.23.0
 # Remove the <=1.10.1 dependency below once sklearn's issue is fixed. The higher version causes
-# issues with sklear's _most_frequent() function using scipy's mode() function
+# issues with sklearn's _most_frequent() function using scipy's mode() function
 scipy>=1.4.1, <=1.10.1
 tqdm>=4.41.0
 seaborn>=0.11.0

--- a/requirements/requirements.txt
+++ b/requirements/requirements.txt
@@ -26,7 +26,7 @@ importlib_metadata>=1.4; python_version < '3.8'
 statsmodels>=0.11.0; python_version < '3.7'
 statsmodels>=0.13.5; python_version >= '3.7'
 # Remove the <=1.10.1 dependency below once sklearn's issue is fixed. The higher version causes
-# issues with sklear's _most_frequent() function using scipy's mode() function
+# issues with sklearn's _most_frequent() function using scipy's mode() function
 scipy>=1.4.1, <=1.10.1
 dataclasses>=0.6; python_version < '3.7'
 plotly>=5.13.1

--- a/requirements/requirements.txt
+++ b/requirements/requirements.txt
@@ -25,7 +25,7 @@ importlib_metadata>=1.4; python_version < '3.8'
 # is updated, explicitly add it here
 statsmodels>=0.11.0; python_version < '3.7'
 statsmodels>=0.13.5; python_version >= '3.7'
-scipy>=1.4.1
+scipy>=1.4.1, <=1.10.1;
 dataclasses>=0.6; python_version < '3.7'
 plotly>=5.13.1
 matplotlib>=3.3.4

--- a/requirements/requirements.txt
+++ b/requirements/requirements.txt
@@ -25,6 +25,8 @@ importlib_metadata>=1.4; python_version < '3.8'
 # is updated, explicitly add it here
 statsmodels>=0.11.0; python_version < '3.7'
 statsmodels>=0.13.5; python_version >= '3.7'
+# Remove the <=1.10.1 dependency below once sklearn's issue is fixed. The higher version causes
+# issues with sklear's _most_frequent() function using scipy's mode() function
 scipy>=1.4.1, <=1.10.1
 dataclasses>=0.6; python_version < '3.7'
 plotly>=5.13.1

--- a/requirements/requirements.txt
+++ b/requirements/requirements.txt
@@ -25,7 +25,7 @@ importlib_metadata>=1.4; python_version < '3.8'
 # is updated, explicitly add it here
 statsmodels>=0.11.0; python_version < '3.7'
 statsmodels>=0.13.5; python_version >= '3.7'
-scipy>=1.4.1, <=1.10.1;
+scipy>=1.4.1, <=1.10.1
 dataclasses>=0.6; python_version < '3.7'
 plotly>=5.13.1
 matplotlib>=3.3.4

